feat(website): add localization ux guidelines page (#2354)
twilio-labs/paste: packages/paste-website/src/pages/foundations/localization/index.mdx

---
title: UX Localization Guidelines

> description: Localization stands for adapting a product, service or content to fit a specific market or country.

Reviewer: suggestion: I would use "refers to" here as opposed to "stands for" here since localization is not an acronym or abstracted piece of language.
```suggestion
description: Localization refers to adapting a product, service or content to fit a specific market or country.
```

Author: Yeah, that is a much better word to use here. Thank you!

Author: Added!

import {Button} from '@twilio-paste/button';
import {Anchor} from '@twilio-paste/anchor';
import {Stack} from '@twilio-paste/stack';
import {DeleteIcon} from '@twilio-paste/icons/esm/DeleteIcon';

import {Callout, CalloutTitle, CalloutText} from '../../../components/callout';
import {DoDont, Do, Dont} from '../../../components/DoDont';

import {SidebarCategoryRoutes} from '../../../constants';

export const pageQuery = graphql`
  {
    mdx(fields: {slug: {eq: "/foundations/localization/"}}) {
      fileAbsolutePath
      frontmatter {
        slug
        title
      }
      headings {
        depth
        value
      }
    }
  }
`;

<content>

<ComponentHeader
  name={props.pageContext.frontmatter.title}
  description={props.pageContext.frontmatter.description}
  categoryRoute={SidebarCategoryRoutes.FOUNDATIONS}
/>

</content>

---

<contentwrapper>

<PageAside data={props.data.mdx} />

<content>

## Introduction

Self-service tips on how to build experiences in a way that’s appropriate and meaningful to all our customers, no matter where they are in the world.

### Definition of terms

<h4 aria-describedby="internationalization">Internationalization</h4>

> <p id="internationalization">

Reviewer 2: Question: why are you adding the HTML tags and id's manually? Usually we just let markdown handle it all for us.

Author: Oh I didn't remove the `aria-describedby` attributes 🤦‍♀️ will do now.

I only added an `id` to the paragraph elements because I wanted to add the attribute to the headers.

Reviewer 2: Ok, cool! Yeah its not necessary to do that.

</p>
<h4 aria-describedby="internationalization">Localization</h4>
<p id="internationalization">
  Localization refers to the adaptation of a product or interface to meet the language, cultural and other requirements
  of a specific target market/locale.
</p>
<h4 aria-describedby="internationalization">Translation</h4>
<p id="internationalization">
  Converting text from one language to another. Not to be confused with localization, translation is just one part of
  localizing a product.
</p>
<h4 aria-describedby="internationalization">Locale</h4>
<p id="internationalization">
  A linguistic region defined by both its language and country. Please note that languages and countries won’t always
  have a 1:1 mapping.
</p>

| Linguistic region | Language   | Country |
| ----------------- | ---------- | ------- |
| fr-FR             | French     | France  |
| fr-CA             | French     | Canada  |
| pt-Brazil         | Portuguese | Brazil  |

<br></br>

## General guidelines

Continue reading to learn how to build experiences in a way that’s appropriate and meaningful to all our customers, no matter where they are in the world.

### Leave blank space around condensed UI components

> Leave blank space around condensed UI components, such as popovers, buttons, and tabs to prevent unexpected line breaks.

Reviewer 3: ```suggestion
- Leave blank space around condensed UI components, such as popovers, buttons, and tabs to prevent unexpected line breaks.
```

Author: Added!

> - For some languages, text is up to 30% longer on average than English.

Reviewer 3: ```suggestion
  - For some languages, text is up to 30% longer on average than English.
```

Author: Added!

- If text gets truncated or has line-breaks at wrong places could lead to grammatical errors and design breaks.
- Make all UI components expandable (auto-layout).

> - Don’t assign fixed-width or height to UI components, unless there’s a strong.

Author: Added!
